src/thor/astar_bss.cc

```diff
@@ -42,10 +44,9 @@ AStarBSSAlgorithm::~AStarBSSAlgorithm() {
 // Clear the temporary information generated during path construction.
 void AStarBSSAlgorithm::Clear() {
   // Reduce edge labels capacity if it's more than limit
-  if (edgelabels_.size() > max_reserved_labels_count_) {
-    edgelabels_.resize(max_reserved_labels_count_);
-    edgelabels_.shrink_to_fit();
-  }
+  edgelabels_.resize(clear_reserved_memory_ ? 0 : max_reserved_labels_count_);
+  edgelabels_.shrink_to_fit();
```

Comment:
my suggestion was wrong here. when you dont have clear_reserved_memory_ set to true and the edges labels container is smaller than max_reserved_labels_count_ then you dont actually have to resize and shrink_to_fit.\



```c++
auto reservation = clear_reserved_memory_ ? 0 : max_reserved_labels_count_; // do this in the constructor and save it as a member variable?
if (edgelabels_forward_.size() > reservation) {
    edgelabels_forward_.resize(reservation);
    edgelabels_forward_.shrink_to_fit();
}

```

Reply:
i tried fixing this in bidirectional a* and dijkstras but its still slow... where else are we running into this, it clearly has to be this bit of the code that is causign the performane bottleneck. i'll  look at it with fresh eyes in a day or so

Reply:
@kevinkreiser 
you haven't watched this PR again?